### PR TITLE
manual pub package roll; remove test_api pin, unblock bumping dwds

### LINKED ISSUE #106886: [flutter_tools] the tool manually pinning test_api is blocking rolling the latest dwds
To diagnose why dwds is did not get bumped by the auto-roll: https://github.com/flutter/flutter/pull/106837, I temporarily added `'dwds': '15.0.0',` to the manual pin map: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/update_packages.dart#L29, and ran `flutter update-packages --force-upgrade` and got this error from pub:

```
Because test_core >=0.4.14 <0.4.15 depends on test_api 0.4.10 and test_core >=0.4.11 <0.4.14 depends on vm_service >=6.0.0 <9.0.0, test_core >=0.4.11 <0.4.15 requires test_api 0.4.10 or vm_service >=6.0.0 <9.0.0.
And because dwds >=15.0.0 depends on vm_service ^9.0.0, if dwds >=15.0.0 and test_core >=0.4.11 <0.4.15 then test_api 0.4.10.
And because test_core >=0.4.15 depends on test_api 0.4.11 and flutter_update_packages depends on test_api 0.4.9, dwds >=15.0.0 is incompatible with test_core >=0.4.11.
So, because flutter_update_packages depends on both dwds 15.0.0 and test_core >=0.4.13, version solving failed.
```

Because we're also manually pinning `test_api`.

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes https://github.com/flutter/flutter/issues/106886

### CODE REVIEW

**Comment by christopherfujino** [comment]
Ok, I need to fix compilation errors...

**Comment by christopherfujino** [comment]
> Ok, I need to fix compilation errors...

Turns out this was just a single nullable return type -> non-nullable extending a dwds interface.

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -10,10 +10,10 @@ dependencies:
   archive: 3.3.0
   args: 2.3.1
   browser_launcher: 1.1.1
-  dds: 2.2.1
-  dwds: 14.0.3
+  dds: 2.2.5
+  dwds: 15.0.0
```
Comment (christopherfujino) [inline packages/flutter_tools/pubspec.yaml:14]: cc @elliette

**Review by elliette — APPROVED** [review]
LGTM, thanks! @helin24 is anything else needed for https://github.com/dart-lang/webdev/issues/1632 ?

**Review by helin24 — APPROVED** [review]
Thanks for alerting me to this change! @elliette and @annagrin this covers the DWDS part of the change I needed to make.

**Comment by annagrin** [comment]
LGTM!

**Pull request merged**